### PR TITLE
[Spotbugs] Enable spotbugs in pulsar-metadata

### PR DESCRIPTION
**Pull request by RobertIndie**
Fixes streamnative/pulsar#1797

### Motivation

Enable spotbugs in module pulsar-metadata.

### CODE REVIEW

**Review by sijie — APPROVED** [review]

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/GetResult.java [hunk 1]
```diff
@@ -26,11 +26,20 @@
 @Data
 public class GetResult {
 
+    public GetResult(byte[] value, Stat stat){
+        this.value = value.clone();
```
Comment (merlimat) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/GetResult.java:30]: While I understand the immutability aspect, making 2 extra copies and allocations of the value array it's overkill for this scenario.

Comment (RobertIndie) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/GetResult.java:30]: Maybe we could make the `value` public to solve this problem(`EI_EXPOSE_REP`) and avoid copy, even though that would sacrifice immutability. What do you think?

Comment (merlimat) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/GetResult.java]: I'm not super convinced in making the field public. `GetResult` class is part of API so I'd rather keep that behind a getter method. An implementation might not store it directly as a `byte[]` but instead convert that on demand.

Comment (merlimat) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/GetResult.java]: Can we instead ignore the warning for this case?

Comment (RobertIndie) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/GetResult.java]: Okay.

**Review by sijie — APPROVED** [review]

**Pull request merged**